web_access_rule_buttons/static/src/js/form_controller.js

```diff
@@ -0,0 +1,32 @@
+/* Copyright 2016 Camptocamp SA
+ * License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). */
+
+odoo.define("web_access_rule_buttons.main", function (require) {
+    "use strict";
+    var FormController = require("web.FormController");
+    FormController.include({
+        async _update(state, params) {
+            return this._super(state, params).then(this.show_hide_buttons(state));
+        },
```

Comment:
I suggest rewriting this in modern js

```suggestion
        async _update(state, params) {
            const res = await this._super.apply(this, arguments);
            await this.show_hide_buttons(state);
            return res;
        },
```